animal.go

```diff
@@ -27,48 +27,30 @@ type Question struct {
 // StartingQuestion - This is the first Question
 var StartingQuestion = "Does it have 4 legs?"
 
+var mux sync.Mutex
+
 type game struct {
-	Running bool
-	Data    map[string]Question
+	Running    bool
+	Data       map[string]Question
+	transcript strings.Builder
 }
 
 // NewGame - Initializing a new game with Starting Data and Running State
-func NewGame() game {
-	var StartingData = map[string]Question{
-		"Does it have 4 legs?": {
-			Yes: "Does it have stripes?",
-			No:  "Is it carnivorous?",
-		},
-		"Does it have stripes?": {
-			Yes: "Is it a zebra?",
-			No:  "Is it a lion?",
-		},
-		"Is it carnivorous?": {
-			Yes: "Is it a snake?",
-			No:  "Is it a worm?",
-		},
-		"Is it a zebra?": {
-			Yes: AnswerWin,
-			No:  AnswerLose,
-		},
-		"Is it a giraffe?": {
-			Yes: AnswerWin,
-			No:  AnswerLose,
-		},
-		"Is it a lion?": {
-			Yes: AnswerWin,
-			No:  AnswerLose,
-		},
-		"Is it a snake?": {
-			Yes: AnswerWin,
-			No:  AnswerLose,
-		},
-		"Is it a worm?": {
-			Yes: AnswerWin,
-			No:  AnswerLose,
-		},
+func NewGame() *game {
+	var StartingData map[string]Question
+	mux.Lock()
+	content, err := ioutil.ReadFile("./data.json")
+	if err != nil {
+		log.Fatal("Error when opening file: ", err)
+	}
+	mux.Unlock()
```

Comment:
When you do use a mutex, the standard pattern is:

```go
m.Lock()
defer m.Unlock()
```

...otherwise you could end up leaking it, just like a file or a response body.